infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/LogReplicationHandler.java

```diff
@@ -34,39 +32,51 @@
+            .generateHandlers(MethodHandles.lookup(), this)
+            .generateErrorHandlers(MethodHandles.lookup(), this);
+
     /**
      * Handle an ACK from Log Replication server.
      *
-     * @param msg The ack message
-     * @param ctx The context the message was sent under
-     * @param r   A reference to the router
+     * @param response The ack message
+     * @param ctx      The context the message was sent under
+     * @param router   A reference to the router
      */
-    @ClientHandler(type = CorfuMsgType.LOG_REPLICATION_ENTRY)
-    private static Object handleLogReplicationAck(CorfuPayloadMsg<LogReplicationEntry> msg,
-                                                  ChannelHandlerContext ctx, IClientRouter r) {
+    @ResponseHandler(type = PayloadCase.LR_ENTRY_RESPONSE)
+    private static Object handleLogReplicationAck(@Nonnull ResponseMsg response,
+                                                  @Nonnull ChannelHandlerContext ctx,
+                                                  @Nonnull IClientRouter router) {
         log.debug("Handle log replication ACK");
-        return msg.getPayload();
+        return response.getPayload().getLrEntryResponse();
     }
 
-    @ClientHandler(type = CorfuMsgType.LOG_REPLICATION_METADATA_RESPONSE)
-    private static Object handleLogReplicationMetadata(CorfuPayloadMsg<LogReplicationMetadataResponse> msg,
-                                                       ChannelHandlerContext ctx, IClientRouter r) {
+    @ResponseHandler(type = PayloadCase.LR_METADATA_RESPONSE)
+    private static Object handleLogReplicationMetadata(@Nonnull ResponseMsg response,
+                                                       @Nonnull ChannelHandlerContext ctx,
+                                                       @Nonnull IClientRouter router) {
         log.debug("Handle log replication Metadata Response");
-        return msg.getPayload();
+        return response.getPayload().getLrMetadataResponse();
     }
 
-    @ClientHandler(type = CorfuMsgType.LOG_REPLICATION_QUERY_LEADERSHIP_RESPONSE)
-    private static Object handleLogReplicationQueryLeadershipResponse(CorfuPayloadMsg<LogReplicationQueryLeaderShipResponse> msg,
-                                                                      ChannelHandlerContext ctx, IClientRouter r) {
-        log.trace("Handle log replication query leadership response msg {}", msg);
-        return msg.getPayload();
+    @ResponseHandler(type = PayloadCase.LR_LEADERSHIP_RESPONSE)
+    private static Object handleLogReplicationQueryLeadershipResponse(@Nonnull ResponseMsg response,
+                                                                      @Nonnull ChannelHandlerContext ctx,
+                                                                      @Nonnull IClientRouter router) {
+        log.trace("Handle log replication query leadership response msg {}", response);
```

Comment:
Here as well: `shortDebugString` and guard.

Reply:
Done.